src/react/utils.js

```diff
@@ -704,3 +706,31 @@ export function getComponentName(
   }
   return "Unknown";
 }
+
+export function convertConfigObjectToReactComponentTreeConfig(
+  realm: Realm,
+  config: ObjectValue | UndefinedValue
+): ReactComponentTreeConfig {
+  // defaults
+  let serverSideRenderOnly = false;
+
+  if (!(config instanceof UndefinedValue)) {
+    for (let [key] of config.properties) {
+      let propValue = getProperty(realm, config, key);
+      invariant(
```

Comment:
This config is derived from user code, right? If there's any way the values are not a string/number/boolean, then this should be a user-facing diagnostics thingy and not an invariant.